crates/http-api-bindings/src/fastchat.rs

```diff
@@ -0,0 +1,93 @@
+use async_trait::async_trait;
+use reqwest::header;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tabby_inference::{TextGeneration, TextGenerationOptions};
+
+#[derive(Serialize)]
+struct Request {
+    model: String,
+    prompt: Vec<String>,
+    max_tokens: usize,
+    temperature: f32,
+}
+
+#[derive(Deserialize)]
+struct Response {
+    choices: Vec<Prediction>,
+}
+
+#[derive(Deserialize)]
+struct Prediction {
+    text: Vec<String>,
+}
+
+pub struct FastChatEngine {
+    client: reqwest::Client,
+    api_endpoint: String,
+    model_name: String,
+}
+
+impl FastChatEngine {
+    pub fn create(api_endpoint: &str, model_name: &str, authorization: &str) -> Self {
+        let mut headers = reqwest::header::HeaderMap::new();
+        if authorization.len() > 0 {
+            headers.insert(
+                "Authorization",
+                header::HeaderValue::from_str(authorization)
+                .expect("Failed to create authorization header"),
+                );
+        }
+        let client = reqwest::Client::builder()
+            .default_headers(headers)
+            .build()
+            .expect("Failed to construct HTTP client");
+        Self {
+            api_endpoint: api_endpoint.to_owned(),
+            model_name: model_name.to_owned(),
+            client,
+        }
+    }
+
+    pub fn prompt_template() -> String {
+        "{prefix}<MID>{suffix}".to_owned()
+    }
+}
+
+#[async_trait]
+impl TextGeneration for FastChatEngine {
+    async fn generate(&self, prompt: &str, options: TextGenerationOptions) -> String {
+        let _stop_sequences: Vec<String> = options
+            .stop_words
+            .iter()
+            .map(|x| x.to_string())
+            .collect();
+
+        let tokens: Vec<&str> = prompt.split("<MID>").collect();
+        let request = Request {
+            model: self.model_name.to_owned(),
+            prompt: vec![tokens[0].to_owned()],
```

Comment:
can we pass toklens[1] to `suffix`?

Reply:
For fastchat v1/completions(https://github.com/lm-sys/FastChat/blob/main/fastchat/serve/openai_api_server.py#L478), it deems prompt as a string. I am not sure how tabby would use suffix, just concat prefix+suffix together?